compendium/DeclarativeServices/src/manager/BindingPolicyDynamicReluctant.cpp

```diff
@@ -61,6 +61,23 @@ namespace cppmicroservices
 
                     notifications.emplace_back(mgr.metadata_.name, RefEvent::REBIND, reference);
                 }
+
+                // for multiple cardinality rebind to new reference if number of
+                // bound references is within limit of maxCardinality value
+                // otherwise log to the user that further bind is not possible
+                if (mgr.IsMultiple()) {
+                    if (mgr.GetBoundReferences().size() < mgr.metadata_.maxCardinality) {
+                        Log("Notify BIND for reference " + mgr.metadata_.name);
+
+                        ClearBoundRefs();
```

Comment:
What if maxCardinaltiy is > 1. Don't want to ClearBoundRefs here? Could be clearing one we want to remain bound?

Reply:
the 'isMultiple' check is checking for maxCardinality > 1 itself.
We clear all bound refs first and then again bind to all possible matches using UpdateBoundRefs below, so we get the references bound before plus new ref.